### PR TITLE
Feature: Google Authentication with Chrome Custom Tabs

### PR DESCRIPTION
**Pull request by akueisara**
### Description

- [x] Implement the GET API request with url query parameters
- [x] Cancel the transition of exiting the VirtusizeWebViewFragment at onStop(), which is called when a user navigates away from the fragment (when a Chrome Custom Tab is opened)
- [x] Move some common code to virtusize-core and fix unit tests

### Demo

Nov-08-2021 17-59-30

### CODE REVIEW

File: virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt [hunk 1]
```diff
@@ -121,7 +123,7 @@ class VirtusizeWebViewFragment : DialogFragment() {
                                     return true
                                 }
                             }
-                            return false
+                            return VirtusizeAuth.isSNSAuthUrl(this@VirtusizeWebViewFragment, url)
```
Comment (akueisara) [inline virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt:126]: Please refer to https://github.com/virtusize/virtusize-auth-android/pull/2

File: virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt [hunk 2]
```diff
@@ -178,6 +180,18 @@ class VirtusizeWebViewFragment : DialogFragment() {
...
+    override fun onStop() {
+        super.onStop()
+        // Cancel the window enter animation
+        dialog?.window?.setWindowAnimations(R.style.VirtusizeDialogFragmentAnimation_Null)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        // Handle the result of the Virtusize SNS auth flow
+        VirtusizeAuth.handleVirtusizeAuthResult(binding.webView, requestCode, resultCode, data)
```
Comment (akueisara) [inline virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt:192]: Please refer to https://github.com/virtusize/virtusize-auth-android/pull/2

File: virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt [hunk 3]
```diff
@@ -179,8 +180,15 @@ class VirtusizeWebViewFragment : DialogFragment() {
...
+    override fun onStop() {
+        super.onStop()
+        // Cancel the window enter animation
+        dialog?.window?.setWindowAnimations(R.style.VirtusizeDialogFragmentAnimation_Null)
+    }
+
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
+        // Handle the result of the Virtusize SNS auth flow
         VirtusizeAuth.handleVirtusizeAuthResult(binding.webView, requestCode, resultCode, data)
```
Comment (akueisara) [inline virtusize/src/main/java/com/virtusize/android/ui/VirtusizeWebViewFragment.kt:192]: Please refer to virtusize/virtusize-auth-android#2

**Review by zr0z — APPROVED** [review]
LGTM!

**Pull request merged**